Ghidra/Features/Base/src/main/java/ghidra/app/util/navigation/GoToQuery.java

```diff
@@ -321,6 +334,109 @@ private boolean processAddressExpression() {
 		return false;
 	}
 
+	private boolean processComponentExpression() {
+		// check for defined data at the currentAddress
+		Program program = navigatable.getProgram();
+		Listing listing = program.getListing();
+		Address address = navigatable.getLocation().getAddress();
+		Data data = listing.getDataAt(address);
+		if (data == null) {
```

Comment:
When converting a program location to the data item the user's cursor is located in, you can't just use the address because of nested data types.

Use DataUtilities.getDataAtLocation(ProgramLocation). That method uses the component path to drill down into nested types if necessary to get the data type the user had focused.

Reply:
I was unaware that `DataUtilities.getDataAtLocation(ProgramLocation)` would go that far. I knew there was away to get the Data containing the location but didn't there was a way to get to exactly where the cursor is. I'm changing this now and will likely move up to the next array/structure if the data at the cursor currently is not one.

I should probably do the same for the one at line 405 too.

Reply:
I created a temporary static method to get the data component directly at the address even if it has a parent. It is probably best to move it into DataUtilities if there isn't a similar one present somewhere else. Just not sure what to name it so that it doesn't conflict with the existing method and has a clear intention. 

https://github.com/NationalSecurityAgency/ghidra/pull/2004/files#diff-e018c4bfdf6af3b238b9f08656efd7d1R383-R389